shared/bytesutil/bytes.go

```diff
@@ -241,3 +241,50 @@ func PadTo(b []byte, size int) []byte {
 	}
 	return append(b, make([]byte, size-len(b))...)
 }
+
+// SetBit sets the index `i` of bitlist `b` to 1.
+// Returns the original bitlist if the index `i`
+// is out of range.
+func SetBit(b []byte, i int) []byte {
+	if i >= len(b)*8 {
+		return b
+	}
+
+	bit := uint8(1 << (i % 8))
+	b[i/8] |= bit
+	return b
+}
+
+// ClearBit clears the index `i` of bitlist `b`.
+// Returns the original bitlist if the index `i`
+// is out of range.
+func ClearBit(b []byte, i int) []byte {
+	if i >= len(b)*8 {
+		return b
+	}
+
+	bit := uint8(1 << (i % 8))
+	b[i/8] &^= bit
+	return b
+}
+
+// MakeEmptyBitfields returns an empty bitlist with
+// input size `i`.
+func MakeEmptyBitfields(i int) []byte {
+	return make([]byte, (i+(8-i%8))/8)
+}
+
+// HighestBitIndex returns the index of the highest
+// bit set from bitlist `b`.
+func HighestBitIndex(b []byte) int {
```

Comment:
What happens if b is nil or empty list?

How do you differentiate `[]byte{0b1}` from empty list? Both of these would return 0

Reply:
Edited it to return error. Thanks for the feedback!